Transform Over Array/code.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @param {number[]} arr
+ * @param {Function} fn
+ * @return {number[]}
+ */
+var map = function(arr, fn) {
+    let res=[]
+    let i=0
+    while(i<=arr.length-1){
+        res[i]=fn(arr[i],i);
+        i++;
+    }
+    return res;
+};
+Updated code with single array
+/**
+ * @param {number[]} arr
+ * @param {Function} fn
+ * @return {number[]}
+ */
+var map = function(arr, fn) {
```

Comment:
never update variables directly received in arguments. Javascript using **call by reference** for Array arguments. so here after calling updated method, your original array who was passed in function would have also updated.